tests/rptest/services/flink.py

```diff
@@ -202,36 +201,64 @@ def __init__(self, context, *args, **kwargs):
         ]
 
-        # Provider instance/node util class
-        nodeutils = ProviderInstanceUtils(self.logger)
-        # Get specs for an flink node
-        specs = nodeutils.get_node_specs(self.nodes[0], context)
-        self.logger.info(f"Using specs of '{specs}'")
-
-        # Handle service configuration for single node
-        # 10% ram - reserved
-        # 10% ram - job manager
-        # 80% ram / vcpus - per taskmanager
-        # Example, xlarge, 32 ram, 4 vcpus
-        # 32 * 0.1 = 3 (rounded down)
-        # 32 * 0.8 / vcpus = 6 (rounded down)
-        jm_ram = int(specs["ram"] * 0.1)
-        tm_ram = int(specs["ram"] * 0.8 // specs["vcpus"])
+        self.logger.info(f"Using specs: {num_cpus} cpus, {node_ram_mb}MB RAM")
         self.service_config = flink_config
-        # default, 1600m
-        self.service_config['jobmanager.memory.process.size'] = f"{jm_ram}g"
-        # default, 1728m
-        self.service_config['taskmanager.memory.process.size'] = f"{tm_ram}g"
-        # 1280m, total - 1g for JVM metadata, etc
-        self.service_config['taskmanager.memory.flink.size'] = f"{tm_ram-1}g"
+
+        tm_count = num_cpus
+        if node_ram_mb < 10240:
+            # Handle service configuration for single node in case docker
+            # I.e. low memory scenarios
+            # 0% ram - no ram reserve needed for docker virtual memory
+            # 792mb ram - job manager
+            # total - 792mb = 1384mb ram - for taskmanager
+            # Tight memory scenario
+            jm_ram = 792
+            tm_ram = int(node_ram_mb - jm_ram // num_cpus)
+            if tm_ram < 1024:
+                tm_count = 1
+                new_tm_ram = node_ram_mb - jm_ram
+                self.logger.info(f"Using {tm_count} taskmanager with "
+                                 f"{new_tm_ram}MB due to low calculated "
+                                 f"memory size ({tm_ram}MB)")
+                tm_ram = new_tm_ram
+            # use min memory, default case is 792mb out of 2048mb
+            self.service_config[
+                'jobmanager.memory.process.size'] = f"{jm_ram}m"
+            # default, ~1280mb out of 2048 if 1 cpu
+            self.service_config[
+                'taskmanager.memory.process.size'] = f"{tm_ram}m"
+            # 1000mb, total - 792mb for JVM metadata, etc
+            self.service_config[
+                'taskmanager.memory.flink.size'] = f"{tm_ram - 512}m"
+        else:
+            # Handle service configuration for single node in case of EC2
+            # 10% ram - reserved
+            # 10% ram - job manager
+            # 80% ram / vcpus - per taskmanager
+            # Example, xlarge, 32 ram, 4 vcpus
+            # 32 * 0.1 = 3 (rounded down)
+            # 32 * 0.8 / vcpus = 6 (rounded down)
+            node_ram_gb = node_ram_mb // 1024
+            jm_ram = int(node_ram_gb * 0.1)
+            tm_ram = int(node_ram_gb * 0.8 // num_cpus)
+            # default, 1600m
```

Comment:
Is this comment right?

Reply:
yep. Default value is ~1600m in flink. But is will not work if you do not set any one out of 3. It's Java and it needs a starting point. :)